File: mycollab-scheduler/src/main/java/com/esofthead/mycollab/schedule/email/format/DateFieldFormat.java
package com.esofthead.mycollab.schedule.email.format;

import java.util.Date;

import com.esofthead.mycollab.core.utils.DateTimeUtils;
import com.hp.gagawa.java.elements.Span;

/**
 * 
 * @author MyCollab Ltd.
 * @since 4.0
 * 
 */
public class DateFieldFormat extends FieldFormat<Date> {

	public DateFieldFormat(String displayName) {
		super(displayName);
	}

	@Override
	public String formatField(Date value, String timeZone) {
		if (value == null)
			return new Span().write();

		return new Span().appendText(
				DateTimeUtils.converToStringWithUserTimeZone(value, timeZone))
				.write();
	}
}

File: mycollab-scheduler/src/main/scala/com/esofthead/mycollab/schedule/email/format/DateFieldFormat.scala
package com.esofthead.mycollab.schedule.email.format

import java.util.Date

import com.esofthead.mycollab.core.utils.{DateTimeUtils, StringUtils, TimezoneVal}
import com.esofthead.mycollab.schedule.email.MailContext
import com.hp.gagawa.java.elements.Span
import org.apache.commons.beanutils.PropertyUtils
import org.slf4j.LoggerFactory

/**
  * @author MyCollab Ltd.
  * @since 4.6.0
  */
class DateFieldFormat(fieldName: String, displayName: Enum[_]) extends FieldFormat(fieldName, displayName) {
  private val LOG = LoggerFactory.getLogger(classOf[DateFieldFormat])

  override def formatField(context: MailContext[_]): String = {
    val wrappedBean = context.wrappedBean
    var value: AnyRef = null
    try {
      value = PropertyUtils.getProperty(wrappedBean, fieldName)
      if (value == null) {
        new Span().write
      } else {
        new Span().appendText(DateTimeUtils.formatDate(value.asInstanceOf[Date], context.user.getDateFormat, TimezoneVal.valueOf(context.getUser.getTimezone))).write
      }
    } catch {
      case e: Any =>
        LOG.error("Can not generate email field: " + fieldName, e)
        new Span().write
    }
  }

  override def formatField(context: MailContext[_], value: String): String = {
    if (StringUtils.isBlank(value)) {
      new Span().write
    }

    DateTimeUtils.convertToStringWithUserTimeZone(value, context.user.getDateFormat, context.getTimeZone)
  }
}
